src/collider/BVHTree.cpp

```diff
@@ -158,3 +160,252 @@ BVHNode *BVHTree::AllocNode()
 		Log::Error("Out of space in m_bvhNodes.");
 	return &m_bvhNodes[m_nodeAllocPos++];
 }
+
+double BVHTree::CalculateSAH() const
+{
+	double outSAH = 0.0;
+	std::vector<BVHNode *> m_nodeStack = { m_root };
+
+	while (!m_nodeStack.empty()) {
+		BVHNode *node = m_nodeStack.back();
+		m_nodeStack.pop_back();
+
+		if (!node->IsLeaf() && node->kids[0])
+			m_nodeStack.push_back(node->kids[0]);
+		if (!node->IsLeaf() && node->kids[1])
+			m_nodeStack.push_back(node->kids[1]);
+
+		// surface area = 2 * width * depth * height
+		vector3d size = node->aabb.max - node->aabb.min;
+		double area = 2.f * size.x * size.y * size.z;
```

Comment:
Odd that we have a `2.f` being multiplied by a double here, would make more sense for it to just be `2.0`

Reply:
Copy-paste / refactoring error, thanks!